src/Gui/DlgSettingsWorkbenchesImp.cpp

```diff
@@ -0,0 +1,475 @@
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+#ifndef _PreComp_
+#include <QCheckBox>
+#include <QPushButton>
+#include <sstream>
+#endif
+
+#include "DlgSettingsWorkbenchesImp.h"
+#include "ui_DlgSettingsWorkbenches.h"
+#include "Application.h"
+#include "UserSettings.h"
+#include "Workbench.h"
+#include "WorkbenchManager.h"
+
+
+using namespace Gui::Dialog;
+
+const QString DlgSettingsWorkbenchesImp::iconLabelStr = QString::fromLatin1("iconLabel");
+const QString DlgSettingsWorkbenchesImp::nameLabelStr = QString::fromLatin1("nameLabel");
+const QString DlgSettingsWorkbenchesImp::loadLabelStr = QString::fromLatin1("loadLabel");
+const QString DlgSettingsWorkbenchesImp::loadButtonStr = QString::fromLatin1("loadButton");
+const QString DlgSettingsWorkbenchesImp::enableCheckboxStr = QString::fromLatin1("enableCheckbox");
+const QString DlgSettingsWorkbenchesImp::autoloadCheckboxStr = QString::fromLatin1("autoloadCheckbox");
+
+/* TRANSLATOR Gui::Dialog::DlgSettingsWorkbenchesImp */
+
+/**
+ *  Constructs a DlgSettingsWorkbenchesImp
+ */
+DlgSettingsWorkbenchesImp::DlgSettingsWorkbenchesImp( QWidget* parent )
+    : PreferencePage( parent )
+    , ui(new Ui_DlgSettingsWorkbenches)
+{
+    ui->setupUi(this);
+    connect(ui->AutoloadModuleCombo, QOverload<int>::of(&QComboBox::activated), this, [this](int index) { onStartWbChangedClicked(index); });
+}
+
+/**
+ *  Destroys the object and frees any allocated resources
+ */
+DlgSettingsWorkbenchesImp::~DlgSettingsWorkbenchesImp()
+{
+}
+
+
+void DlgSettingsWorkbenchesImp::saveSettings()
+{
+    std::ostringstream enabledStr, disabledStr, autoloadStr;
+
+    for (int i = 0; i < ui->wbList->count(); i++) {
+        QWidget* widget = ui->wbList->itemWidget(ui->wbList->item(i));
+        if (!widget)
+            continue;
+        QCheckBox* enableBox = widget->findChild<QCheckBox*>(enableCheckboxStr);
+        if (!enableBox)
+            continue;
```

Comment:
This looks super hacky. IMO better to define our own item model with appropriate methods.

Reply:
It was how it was done before for the load and auto-load checkboxes. I reused the same pattern here.
But yes it's true I can make a custom widget class with pointers to the different subwidgets. It would make more sense.

Reply:
You don't need pointers. You can use a customized QListView, or QItemDelegate to handle data.

Reply:
You mean as it's done in Element widget?
I know custom delegate are the best way to do widgets in list. But this is a lot of work that I don't think is justified here because we have a limited number of entries (vs element which can have hundreds of entries). Besides the custom delegate seems harder to maintain as can attest the recent checkbox addition to element widget which have (had?) selection issues. Besides this is not a core functionality that needs to be as efficient as possible.

So I suggest just making a custom widget to access the subwidget in a cleaner way than with the qobject names.

Reply:
It's a bit tweaky regarding the situation, but you can also have a look at `QListWidget::setItemWidget`. Should be enough to do this.

Reply:
setItemWidget is what I'm using now actually. I just forced push with changeds. I made a subclass of qwidget. Can you please check if it is acceptable like this?
Thanks